modules/porous_flow/src/dirackernels/PorousFlowPointEnthalpySourceFromPostprocessor.C

```diff
@@ -0,0 +1,56 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PorousFlowPointEnthalpySourceFromPostprocessor.h"
+#include "SinglePhaseFluidProperties.h"
+
+registerMooseObject("PorousFlowApp", PorousFlowPointEnthalpySourceFromPostprocessor);
+
+InputParameters
+PorousFlowPointEnthalpySourceFromPostprocessor::validParams()
+{
+  InputParameters params = DiracKernel::validParams();
+  params.addRequiredParam<PostprocessorName>("mass_flux",
+                                             "The postprocessor name holding the mass flux at this "
+                                             "point in kg/s (only positive values are allowed)");
+  params.addRequiredParam<UserObjectName>("fp", "The name of the user object for fluid properties");
+  params.addRequiredCoupledVar("pressure", "Pressure");
+  params.addRequiredParam<PostprocessorName>(
+      "T_in", "The postprocessor name holding the temperature of fluid (measured in K)");
+  params.addRequiredParam<Point>("point", "The x,y,z coordinates of the point source");
+  params.addClassDescription(
+      "Point source that adds fluid at a mass flux rate (specified by a postprocessor) and given "
```

Comment:
Point source that adds heat energy corresponding to injection of a fluid with specified mass flux rate (specified by a postprocessor) at given temperature (specified by a postprocessor)